internal/router/router.go

```diff
@@ -734,7 +737,7 @@ func splitId(path string) (url *charm.Reference, rest string, err error) {
 	urlStr := strings.TrimSuffix(path[0:i], "/")
 	url, err = charm.ParseReference(urlStr)
 	if err != nil {
-		return nil, "", errgo.Mask(err)
+		return nil, "", errgo.WithCausef(err, params.ErrNotFound, "")
```

Comment:
I'm not sure the NotFound is an appropriate error for this function to return - "not found" doesn't really make sense as a return from splitting an id.

I'd use WithCausef above rather than Mask and add the not-found status in the caller, where the not-found status makes sense.